keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,392 @@
+namespaceSelector, specifically with a `notIn` operator as shown below, will
+inadvertently select this namespace instead of excluding it, which is an undesired
+side effect.
+
+```
+namespaceSelector:
+  matchExpressions:
+  - key: kubernetes.io/metadata.name
+    operator: NotIn
+    values:
+    - kube-system
+```
+
+Hence, it must be noted that namespaces are not guaranteed to have this label on
+downgrades to release which do not support this feature. Users may choose to label
+their namespaces by themselves in order to restore their expected behaviour in
+terms of network policy rules.
+
+### Version Skew Strategy
+N/A
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+1.21:
+- gate disabled by default
+- if gate disabled, log or event incompatible uses of this
+- if gate enabled, enforce
+- release note
+
+1.22:
+- gate enabled by default
+- release note
+
+1.23:
+- lock gate enabled
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  TBD
+
+* **What specific metrics should inform a rollback?**
+  TBD
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  TBD
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  Even if applying deprecation policies, they may still surprise some users.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+By running `kubectl get ns --show-labels` and inspecting the `kubernetes.io` values.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine** 
+
+Since this just an immutable field, we don't expect any such indicators to be of relevance.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+Since this just an immutable field, we don't expect any such indicators to be of relevance.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+
+Since this just an immutable field, we don't expect any such indicators to be of relevance.
+
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  None
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+
+No, because it is just mutating the incoming created namespaces at the APIServer level.
+
+* **Will enabling / using this feature result in introducing new API types?**
+
+No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**:
+
+No
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+
+No
+
```

Comment:
I think this will actually increase the resource size of Namespace objects and their label index (but probably not noticably).

Reply:
done